pkg/api/datasources.go

```diff
@@ -598,19 +606,27 @@ func getEncodedString(jsonData *simplejson.Json, key string) string {
 	return string(val)
 }
 
-func checkTeamHTTPHeaderPermissions(hs *HTTPServer, c *contextmodel.ReqContext, ds *datasources.DataSource, cmd datasources.UpdateDataSourceCommand) (bool, error) {
+func checkTeamHTTPHeadersDiff(ds *datasources.DataSource, jsonData *simplejson.Json) bool {
```

Comment:
we really need to make sure to test this diff, it is takes a long time to perform or not